niceman/resource/ssh.py

```diff
@@ -169,13 +279,27 @@ def get(self, src_path, dest_path, uid=-1, gid=-1):
         dest_dir, dest_basename = os.path.split(dest_path)
         if not os.path.exists(dest_dir):
             os.makedirs(dest_dir)
-        sftp = self.ssh.open_sftp()
+        sftp = paramiko.SFTPClient.from_transport(self.transport)
         sftp.get(src_path, dest_path)
         sftp.close()
 
         if uid > -1 or gid > -1:
             self.chown(dest_path, uid, gid, remote=False)
 
+    @borrowdoc(POSIXSession)
+    def exists_command(self, path):
+        return ['bash', '-c', '"test', '-e', path, '&&', 'echo', 'Found"']
+
+    @borrowdoc(POSIXSession)
+    def isdir_command(self, path):
+        return ['bash', '-c', '"test', '-d', path, '&&', 'echo', 'Found"']
```

Comment:
This class, `SSHSession`, inherits from `POSIXSession`, and that already defines `exists_command` and `isdir_command` (as well as `get_mtime_command` below), so isn't overriding them here unnecessary?  I didn't spot any differences in the intended command, though I do think the variants defined here will always fail with a bash syntax error.

Reply:
It's not the command that is troublesome but the *double-quotes* for the test command run in the bash shell. There is a difference in how the resulting command after the -c is passed to the execute_command() method which, in turn, is handled slightly differently by the docker, runner, and paramiko packages. I was unable to discover a common quoting pattern that all 3 execution handlers would handle without error in the remote shell.

Reply:
Ah, I see.  Thanks for the explanation.